### PR TITLE
spyglass: Add support for custom title and help info in HTML lens

### PR DESCRIPTION
**Pull request by stbenjam**
This extracts the html `<title>` tag as an alternative to the filename
for the HTML lens, if present. It also allows optionally specifying help
by using a `meta` tag:

  `<meta name="description" content="Custom tools contains links to Loki and Prometheus"/>`

If the HTML doesn't contain a title or meta tag, the existing behavior is maintained.

image

### CODE REVIEW

**Comment by stbenjam** [comment]
/assign alvaroaleman @petr-muller 

Could one of you take a look? Thanks!

File: prow/spyglass/lenses/html/html.go
Comment (petr-muller) [inline prow/spyglass/lenses/html/html.go:154]: Does this behave correctly with empty tag, like `<title></title>` (do we get the `TextToken` with empty string as data so that we can clear `isTitle`)

Comment (stbenjam) [reply prow/spyglass/lenses/html/html.go]: Good point, it'll skip an empty title now

**Review by petr-muller — APPROVED** [review]
/hold
LGTM but I'd prefer if someone with more spyglass/lens experience took a look, too

**Review by alvaroaleman — APPROVED** [review]
Seems fine to me but I am not a frontend expert :upside_down_face:

File: prow/spyglass/lenses/html/html.go
Comment (stbenjam) [reply prow/spyglass/lenses/html/html.go:167]: Performance seems to be about the same, although this version seems to be slightly faster (my guess is string concat is slightly faster than sprintf used in the old version). 


```
BenchmarkStringConcat-12    	 1000000	      1102 ns/op
BenchmarkByteSprintf-12     	 1000000	      1176 ns/op
```

**Comment by stbenjam** [comment]
/hold cancel

Since Alvaro and Petr both signed off I'm going to remove the hold.

**Pull request merged**
